core/integration/telemetry_test.go

```diff
@@ -42,3 +43,27 @@ func (TelemetrySuite) TestInternalVertexes(ctx context.Context, t *testctx.T) {
 		require.NotContains(t, logs.String(), "merge (")
 	})
 }
+
+func (TelemetrySuite) TestMetrics(ctx context.Context, t *testctx.T) {
```

Comment:
This test is flaky in CI, I believe just due to the fact that all the metric plumbing is inherently time based and thus sometimes slower in some test runs.

Best option is probably to rebase on https://github.com/dagger/dagger/pull/8442 and use the new telemetry test setup there.